### PR TITLE
Add a data migration guide.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Adding required links is a good use-case for data migrations.

### CODE REVIEW

**Review by elprans — APPROVED** [review]

**Comment by colinhacks** [comment]
I like it! Is this intended to replace the "change a property into a link" guide? That one always seemed a bit out of place and oddly specific, I think this example covers the same principles/concepts with a more common use case.

File: docs/guides/migrations/reqlink.rst [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+This example shows how to setup a required link. We'll use a
+character in an adventure game as the type of data we will evolve.
+
+Let's start with this schema:
+
+.. code-block:: sdl
+
+    type Character {
+        required property name -> str;
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/guides/migrations/reqlink.rst]: I've updated this guide as well as SDL in other migration guides  to use 2-space indenting.

File: docs/guides/migrations/reqlink.rst [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+.. code-block:: bash
+
+    $ edgedb migration create
+    did you make link 'class' of object type 'default::Character' required?
+    [y,n,l,c,b,s,q,?]
+    > y
+    Please specify an expression to populate existing objects in order to
+    make link 'class' of object type 'default::Character' required:
+    fill_expr> <CharacterClass>{}
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/guides/migrations/reqlink.rst]: Yeah, that's a better expression.

File: docs/guides/migrations/reqlink.rst [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    proceed you must fix the statement to read as:
+      CREATE MIGRATION m1juin65wriqmb4vwg23fiyajjxlzj2jyjv5qp36uxenit5y63g2iq
+      ONTO ...
+    if this migration is not applied to any database. Alternatively,
+    revert the changes to the file.
+
+The migration tool detected that we've altered the file and asks us to
+update the migration name (acting as a checksum) if this was
+deliberate. This is done as a precaution against accidental changes.
+Since we've done this on purpose, we can update the file and run
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/guides/migrations/reqlink.rst:79]: Added diff

File: docs/guides/migrations/reqlink.rst [hunk 4]
```diff
@@ -0,0 +1,322 @@
+.. _ref_migration_reqlink:
+
+==========================
+Setting up a required link
```
[the remark this answers is not in the document]
Comment (vpetrovykh) [reply docs/guides/migrations/reqlink.rst]: Updated.

**Comment by vpetrovykh** [comment]
At the moment this is not replacing the "change a property into a link" guide as I think they follow different internal logic of why you'd do things in a certain way. They do share a lot of the stuff, so maybe this can be seen as the more generic one.

**Pull request merged**